src/linter/rules/no_unresolved.zig

```diff
@@ -0,0 +1,84 @@
+const std = @import("std");
+const fs = std.fs;
+const path = std.fs.path;
+const _source = @import("../../source.zig");
+
+const Ast = std.zig.Ast;
+const Node = Ast.Node;
+const Loc = std.zig.Loc;
+const Span = _source.Span;
+const LinterContext = @import("../lint_context.zig");
+const Rule = @import("../rule.zig").Rule;
+const NodeWrapper = @import("../rule.zig").NodeWrapper;
+
+pub const NoUnresolved = struct {
+    pub const Name = "no-unresolved";
+
+    pub fn runOnNode(_: *const NoUnresolved, wrapper: NodeWrapper, ctx: *LinterContext) void {
+        const node = wrapper.node;
+        if (ctx.source.pathname == null) return; //   anonymous source file
+        if (node.tag != .builtin_call_two) return; // not a node we care about
+
+        const builtin_name = ctx.ast().tokenSlice(node.main_token);
+        if (!std.mem.eql(u8, builtin_name, "@import")) {
+            return;
+        }
+        if (node.data.lhs == 0) {
+            ctx.diagnostic(
+                "Call to `@import()` has no file path or module name.",
+                .{ctx.spanN(wrapper.idx)},
+            );
+        }
+
+        const tags: []Ast.Node.Tag = ctx.ast().nodes.items(.tag);
+        const main_tokens = ctx.ast().nodes.items(.main_token);
+
+        // importing with a constant or something. Skip analysis. Maybe in the
```

Comment:
zig doesnt actually support imports to non-string literals. they require this so their first pass can resolve all files upfront before their semantic analysis begins.

the code is right but the comment can be edited

Reply:
ill make another pr with this added